src/network/C4NetIO.cpp

```diff
@@ -1769,20 +1814,21 @@ bool C4NetIOSimpleUDP::InitBroadcast(addr_t *pBroadcastAddr)
 	}
 
 	// set mc ttl to somewhat about "same net"
-	int iTTL = 16;
-	if (setsockopt(sock, IPPROTO_IP, IP_MULTICAST_TTL, reinterpret_cast<char*>(&iTTL), sizeof(iTTL)) == SOCKET_ERROR)
+	int TTL = 16;
+	if (setsockopt(sock, IPPROTO_IPV6, IPV6_MULTICAST_HOPS, reinterpret_cast<char*>(&TTL), sizeof(TTL)) == SOCKET_ERROR)
 	{
 		SetError("could not set mc ttl", true);
 		return false;
 	}
 
 	// set up multicast group information
 	this->MCAddr = *pBroadcastAddr;
-	MCGrpInfo.imr_multiaddr = static_cast<sockaddr_in*>(&MCAddr)->sin_addr;
-	MCGrpInfo.imr_interface.s_addr = INADDR_ANY;
+	MCGrpInfo.ipv6mr_multiaddr = static_cast<sockaddr_in6>(MCAddr).sin6_addr;
```

Comment:
Ah yes. I don't think that code is very useful in any case. Unicast works fine over the internet, so it is without doubt good enough for the local network. I left the code over there because maybe someone wants to have some fun with IPv6 multicast at some point. It seems to fail silently somewhere, but I can add a check here in any case.